syncd/syncd_applyview.cpp

```diff
@@ -1330,6 +1350,27 @@ class AsicView
                     continue;
                 }
 
+                /*
+                 * NOTE: When operation is SET on OID object it can release
+                 * references on that OID, and if that OID is later to be
+                 * removed a wrong order of operations will happen:
+                 *
+                 * not optimized scenario:
+                 * - create object A
+                 * - set object B attr (releases reference on C)
+                 * - remove object D (release reference on C)
+                 * - remove object C
+                 *
+                 * this logic could result with wrong order:
+                 * - remove D
+                 * - remove C (will break reference count on C, since used in B)
+                 *   since D is considered last OP releasing C reference
+                 * - create A
+                 * - set object B with A
+                 *
+                 * This is fixed by checking if last operation to release reference was REMOVE
+                 */
```

Comment:
If i understand correctly, with your fix, the optimized order is as follows?

remove D
create A
set object B with A
remove C